test/translate_c.zig

```diff
@@ -1764,12 +1764,25 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
         \\pub export var array: [100]c_int = .{0} ** 100;
         \\pub export fn foo(arg_index: c_int) c_int {
         \\    var index = arg_index;
-        \\    return array[index];
+        \\    return array[@intCast(c_uint, index)];
```

Comment:
Can you add a test case where the index is unsigned, and so it can avoid the `@intCast`?

Reply:
:+1:  I already one in my local branch.